File: unit_04/w10_d1/lectures/calculatorChall/js/Calculator.js
class Calculator {
  add(x, y = this.previousAnswer) {
    this.previousAnswer = x + y;
    return this.previousAnswer;
  }

  subtract(x, y) {
    if (!y && this.previousAnswer) {
      this.previousAnswer = this.previousAnswer - x;
    } else {
      this.previousAnswer = x - y;
    }
    return this.previousAnswer;
  }

  multiply(x, y = this.previousAnswer) {
    this.previousAnswer = x * y;
    return this.previousAnswer;
  }

  divide(x, y) {
    if (!y && this.previousAnswer) {
      this.previousAnswer = this.previousAnswer / x;
    } else {
      this.previousAnswer = x / y;
    }
    return this.previousAnswer;
  }
}

module.exports = Calculator;

File: unit_04/w10_d1/lectures/calculatorChall/ruby/calculator.rb
class Calculator
  def initialize
    @previous_answer = 0
  end

  def add(x, y = @previous_answer)
    @previous_answer = x + y
  end

  def subtract(x = @previous_answer, y = 0)
    @previous_answer = x - y
  end

  def subtract(x, y = nil)
    @previous_answer = if !y && @previous_answer
                         @previous_answer - x
                       else
                         @previous_answer = x - y
                       end
  end

  def multiply(x, y = @previous_answer)
    @previous_answer = x * y
  end

  def divide(x, y = nil)
    @previous_answer = if !y && @previous_answer
                         @previous_answer / x
                       else
                         @previous_answer = x / y
                       end
  end
end
